forge-core/src/main/java/forge/util/collect/FCollection.java

```diff
@@ -43,12 +44,40 @@ public static <T> FCollection<T> getEmpty() {
     /**
      * The {@link Set} representation of this collection.
      */
-    private final Set<T> set = Sets.newHashSet();
+    private Set<T> SET;
```

Comment:
Just a question, why don't you use the default value anymore? I mean there shouldn't be a time where the set/list is null?

Reply:
Everytime a new cardcollection is created and isn't used because of invalid condition saves a bit of memory. On my local copy when i profile it (Card, player, spellability, etc), unused vars and not initialized if not needed produce good results.

Reply:
Is that the result of a bunch of fields using them as the default value? Like these in `Card.java`?

```java
    private List<GameCommand> leavePlayCommandList = Lists.newArrayList();
    private final List<GameCommand> untapCommandList = Lists.newArrayList();
    private final List<GameCommand> changeControllerCommandList = Lists.newArrayList();
    private final List<GameCommand> unattachCommandList = Lists.newArrayList();
    private final List<GameCommand> faceupCommandList = Lists.newArrayList();
    private final List<GameCommand> facedownCommandList = Lists.newArrayList();
    private final List<GameCommand> phaseOutCommandList = Lists.newArrayList();
    private final List<Object[]> staticCommandList = Lists.newArrayList();
```

Might be worth giving the same lazy-init treatment to other commonly used classes.

Reply:
I refactor the initial lazy init and added a lazy initialization from the article here: https://4comprehension.com/leveraging-lambda-expressions-for-lazy-evaluation-in-java/

Reply:
> @kevlahnota checkout this:
> 
> https://guava.dev/releases/21.0/api/docs/com/google/common/base/Suppliers.html#memoize-com.google.common.base.Supplier- `public static <T> Supplier<T> memoize(Supplier<T> delegate)`

done

Reply:
@Hanmac @Agetian Anything else needed?

Reply:
Can't think of anything at the moment :)